openedx/stanford/djangoapps/auth_lagunita/management/commands/opt_in_to_marketing.py

```diff
@@ -0,0 +1,117 @@
+    help = __doc__
+
+    access_token = None
+    api_url = None
+    client_id = None
+    client_secret = None
+    dry_run = None
+    list_id = None
+    verbosity = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-l',
+            '--list-id',
+            help='Subscribe users to this Marketo list',
+        )
+        parser.add_argument(
+            '-N',
+            '--no-dry-run',
+            action='store_false',
+            dest='dry_run',
+            help='Actually write to the API',
+        )
+        parser.add_argument(
+            '-t',
+            '--access-token',
+            help='Reuse existing access token',
+        )
+        parser.add_argument(
+            '-u',
+            '--api-url',
+            help='Derive Marketo API endpoints from this base URL',
+        )
+        parser.add_argument(
+            '-i',
+            '--client-id',
+            help='Override Client ID for Marketo API access',
+        )
+        parser.add_argument(
+            '-s',
+            '--client-secret',
+            help='Override Client Secret for Marketo API access',
+        )
+
+    def _initialize_settings(self, **kwargs):
+        """
+        Parse and assign settings from CLI
+        """
+        self.access_token = kwargs['access_token']
+        self.api_url = kwargs['api_url'] or settings.MARKETO_URL
+        self.client_id = kwargs['client_id'] or settings.MARKETO_CLIENT_ID
+        self.client_secret = kwargs['client_secret'] or settings.MARKETO_CLIENT_SECRET
+        self.dry_run = kwargs['dry_run']
+        self.list_id = kwargs['list_id'] or settings.MARKETO_LIST_ID
+        self.verbosity = kwargs['verbosity']
+
+    def _set_logging_verbosity(self):
+        """
+        Toggle verobsity via CLI
+        """
+        if not self.verbosity:
+            log.setLevel(logging.ERROR)
+        elif self.verbosity == 1:
+            log.setLevel(logging.INFO)
+        else:
+            log.setLevel(logging.DEBUG)
+
+    def handle(self, *args, **kwargs):
+        self._initialize_settings(**kwargs)
+        self._set_logging_verbosity()
+        api = MarketoApi(
+            self.api_url,
+            self.access_token,
+            self.client_id,
+            self.client_secret,
+        )
+        log.info('User lookup attempted')
+        need_subscribed = Info.objects.need_subscribed()
+        for info in need_subscribed:
+            user = info.user
+            if self.dry_run:
+                log.info("Subscription faked for user=%s", user.email)
+            else:
+                if not api.access_token:
+                    api.refresh_access_token()
+                subscription = api.subscribe(user, self.list_id)
+                if subscription:
+                    info.submitted_marketing_optin = True
+                    info.save()
+            time.sleep(0.5)
+        if len(need_subscribed):
+            log.info('User lookup succeeded')
```

Comment:
sort of a nit, it feels like the semantics of the message here doesn't quite match the code logic, i.e. The _lookup_ succeeded right after the call to `Info.objects.need_subscribed()`. Whereas here, it feels more accurate to say something like `"Successfully made subscription calls for X users"`